packages/connect-query/src/use-query.ts

```diff
@@ -58,14 +58,18 @@ export function useQuery<
     transport: transport ?? transportFromCtx,
     callOptions,
   });
+  const { enabled: baseEnabled, ...baseRest } = baseOptions;
+  const tsOpts = {
+    ...queryOptions,
+    ...baseRest,
+  };
   // The query cannot be enabled if the base options are disabled, regardless of
   // incoming query options.
-  const enabled = baseOptions.enabled && (queryOptions.enabled ?? true);
-  return tsUseQuery({
-    ...queryOptions,
-    ...baseOptions,
-    enabled,
-  });
+  const enabled = baseEnabled ?? queryOptions.enabled;
+  if (enabled !== undefined) {
+    tsOpts.enabled = enabled;
```

Comment:
Much of the complexity of the change came from the fact that Typescript doesn't allow assigning `undefined` to a `field?` type of field, actually I had never run into that yet. Let me know if there's a pattern I'm not aware of.